src/main/java/org/folio/dcb/service/impl/BaseLibraryService.java

```diff
@@ -102,13 +106,31 @@ public void updateTransactionStatus(TransactionEntity dcbTransaction, Transactio
       log.info("updateTransactionStatus:: transaction status transition from {} to {} for the item with barcode {} ",
         ITEM_CHECKED_IN.getValue(), CLOSED.getValue(), dcbTransaction.getItemBarcode());
       updateTransactionEntity(dcbTransaction, requestedStatus);
+    } else if(CANCELLED == requestedStatus) {
+      log.info("updateTransactionStatus:: Cancelling transaction with id: {} for Borrower/Pickup role", dcbTransaction.getId());
+      cancelTransactionRequest(dcbTransaction);
     } else {
       String errorMessage = String.format("updateTransactionStatus:: status update from %s to %s is not implemented", currentStatus, requestedStatus);
       log.warn(errorMessage);
       throw new IllegalArgumentException(errorMessage);
     }
   }
 
+  public void cancelTransactionRequest(TransactionEntity transactionEntity){
+    try {
+      if(circulationService.cancelRequestIfExistOrNull(transactionEntity) == null){
+        cancelTransactionEntity(transactionEntity);
```

Comment:
Is this condition is correct ?
 Looks like the method **cancelRequestIfExistOrNull** will return null only if the getCancellationRequestIfOpenOrNull method returns null. But we need to update the transaction status when the cancellation happens right ?

Reply:
State of transaction will be changed when kafka triggers cancel event from mod-circulation-storage